mobile/src/main/java/org/openhab/habdroid/core/connection/AbstractConnection.java

```diff
@@ -0,0 +1,104 @@
+package org.openhab.habdroid.core.connection;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.AsyncTask;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import org.openhab.habdroid.util.Constants;
+import org.openhab.habdroid.util.MyAsyncHttpClient;
+import org.openhab.habdroid.util.MySyncHttpClient;
+
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.URL;
+import java.util.concurrent.ExecutionException;
+
+public abstract class AbstractConnection implements Connection {
+    private static final String TAG = AbstractConnection.class.getSimpleName();
+
+    private Context ctx;
+    private SharedPreferences settings;
+
+    private int connectionType;
+
+    private MyAsyncHttpClient asyncHttpClient;
+    private MySyncHttpClient syncHttpClient;
+
+    AbstractConnection(Context ctx){
+        this(ctx, PreferenceManager.getDefaultSharedPreferences(ctx));
+    }
+
+    AbstractConnection(Context ctx, SharedPreferences settings) {
+        this.ctx = ctx;
+        this.settings = settings;
+    }
+
+    public MyAsyncHttpClient getAsyncHttpClient() {
+        if (asyncHttpClient == null) {
+            asyncHttpClient = new MyAsyncHttpClient(ctx, ignoreSslHostname(), ignoreCertTrust());
```

Comment:
Ok, that was stupid. The username and password should be setto the client here...

Reply:
Done